.gitmodules

```diff
@@ -118,3 +118,8 @@
 	path = Sming/Libraries/MCP_CAN_lib
 	url = https://github.com/coryjfowler/MCP_CAN_lib.git
 	ignore = dirty
+[submodule "Sming/Components/libsodium/libsodium"]
```

Comment:
Please, place this submodule definition block after esptool2. Also change the fist line to:
`[submodule "libsodium"]`. This way looking at the gitmodules file the developers can easier figure out that this is a submodule that can be used in all architectures and is not an Arduino-style library.

Reply:
Another point. The code at the moment does not compile for the host architecture. You can repeat the following steps to reproduce the issue:

```
cd $SMING_HOME/../samples/HttpServer_FirmwareUpload
echo "COMPONENT_DEPENDS := libsodium" >> component.mk
make SMING_ARCH=Host ENABLE_GDB=1 ENABLE_LWIPDEBUG=1 ENABLE_MALLOC_COUNT=0
```

The compilation errors with:
```
Sming/Components/libsodium/libsodium/src/libsodium/randombytes/sysrandom/randombytes_sysrandom.c:30:31: error: implicit declaration of function ‘syscall’ [-Werror=implicit-function-declaration]
 #   define getrandom(B, S, F) syscall(SYS_getrandom, (B), (int) (S), (F))
                               ^
/home/slavey/dev/esp8266.dev.box/dev/Sming/Sming/Components/libsodium/libsodium/src/libsodium/randombytes/sysrandom/randombytes_sysrandom.c:265:18: note: in expansion of macro ‘getrandom’
         readnb = getrandom(buf, size, 0);
                  ^
cc1: all warnings being treated as errors
```

It would be great if `libsodium` works also on the Host architecture.

Reply:
It was also my intention to make it available to all supported architectures. 
Adding _GNU_SOURCE fix the issue for me, too. However, It would be great if someone can check this out on a Windows host.
Offering a working solution for all possible realizations of the Host "architecture" can get quite involved, as it is not even restricted to x86-based Hosts.